src/prototype_room_creepbuilder.js

```diff
@@ -51,7 +51,7 @@ Room.prototype.spawnCheckForCreate = function() {
     this.memory.queue.shift();
     return false;
   }
-  creep.ttl = creep.ttl || config.creep.queueTtl;
+  creep.ttl = creep.ttl || Math.ceil(config.creep.queueTtl * _.size(CONTROLLER_DOWNGRADE) / this.controller.level);
```

Comment:
Can you please add a comment or extract it in a method with proper naming, so that it is clearer what it does.

I have no clue why to integrate downgrade or controller level.